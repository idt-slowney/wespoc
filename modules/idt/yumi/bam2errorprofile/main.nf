process YUMI_BAM2ERRORPROFILE {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    // container "027151828055.dkr.ecr.us-west-2.amazonaws.com/platform-poc-snakemake:latest"

    input:
    tuple val(meta), path(bam)
    tuple val(meta2), path(bed)
    tuple val(meta3), path(fasta)

    output:
    tuple val(meta), path("*_mpileup_error_rate.txt"), emit: error_rate
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    mpileup.py bam2ErrorProfile --inbam $bam \
        --inbed $bed \
        --genome $fasta \
        --thread $task.cpus \
        $args \
        --outErrorProfile ${prefix}_mpileup_error_rate.txt
    
    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        bam2ErrorProfile: \$(python --version |& sed '1!d ; s/Python //')
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_mpileup_error_rate.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mpileup: \$(python --version |& sed '1!d ; s/Python //')
    END_VERSIONS
    """
}
