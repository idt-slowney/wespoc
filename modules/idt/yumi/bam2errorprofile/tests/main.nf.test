nextflow_process {

    name "Test Process BAM2ERRORPROFILE"
    script "../main.nf"
    process "YUMI_BAM2ERRORPROFILE"

    tag "modules"
    tag "modules_nfcore"
    tag "bam2errorprofile"

    test("Base Test - no additional args") {

        when {
            process {
                """                
                input[0] = Channel.value([
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true)
                ])
                input[1] = Channel.value([
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data['homo_sapiens']['genome']['genome_bed'], checkIfExists: true)
                ])
                input[2] = Channel.value([
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true)
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("Base Test - with args") {
        config "./bam2errorprofile.config"
        when {
            process {
                """                
                input[0] = Channel.value([
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true)
                ])
                input[1] = Channel.value([
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data['homo_sapiens']['genome']['genome_bed'], checkIfExists: true)
                ])
                input[2] = Channel.value([
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true)
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
