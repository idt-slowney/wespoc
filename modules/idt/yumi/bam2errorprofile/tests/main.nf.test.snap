{
    "Base Test - with args": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_mpileup_error_rate.txt:md5,d043978cf48c1141cba53cf5b4e13890"
                    ]
                ],
                "1": [
                    "versions.yml:md5,9defa834e1a9cd6bdfa5193de12102c8"
                ],
                "error_rate": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_mpileup_error_rate.txt:md5,d043978cf48c1141cba53cf5b4e13890"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,9defa834e1a9cd6bdfa5193de12102c8"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-08-21T14:46:59.902775771"
    },
    "Base Test - no additional args": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_mpileup_error_rate.txt:md5,b5f2871cb5fd57408f5b4d3e95ebc5aa"
                    ]
                ],
                "1": [
                    "versions.yml:md5,9defa834e1a9cd6bdfa5193de12102c8"
                ],
                "error_rate": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_mpileup_error_rate.txt:md5,b5f2871cb5fd57408f5b4d3e95ebc5aa"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,9defa834e1a9cd6bdfa5193de12102c8"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-08-21T14:40:30.442653024"
    }
}